src/mpool.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#include "mpool.h"
+
+typedef struct memchunk {
+    struct memchunk *next;
+} memchunk_t;
+
+typedef struct mpool {
+    long chunk_ct;
+    long page_count;
+    size_t chunk_size;
+    char *mempage;
+    struct memchunk *free_chunk_head;
+    struct mpool *next;
+} mpool_t;
+
+static void *get_mmap(long sz)
```

Comment:
Check the file `src/elf.c` for `USE_MMAP`. At present, you can figure out Microsoft Windows counterpart or stop compilation (by placing `#error`).

Portability is an important consideration.